src/Core/PluginConfig.php

```diff
@@ -0,0 +1,109 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         5.1.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Core;
+
+use Cake\Utility\Hash;
+
+/**
+ * PluginConfig contains all available plugins and their config if/how they should be loaded
+ *
+ * @link https://book.cakephp.org/5/en/plugins.html
+ */
+class PluginConfig
```

Comment:
Is this something we can mark as `internal` or do we have scenarios where userland code would need this?

Reply:
well at least the DebugKit would also use this class to display available plugins compared to the loaded plugins. But I am not sure if userland code would need to have access to this.

Reply:
DebugKit using an internal class of the core is fine.

Reply:
the `@internal` phpdoc has now been added to the PluginConfig class